### PR TITLE
[ION-682] spdx file import

### PR DESCRIPTION
**Pull request by struthj2**
- Updates `cmd/project.go` adding methods for `create-spdx-project` command and appropriate flags for spdx import.
  - adds optional `poc-email` flag
  - adds `ext/examples/SPX_USAGE.md` showing example spdx import commands
  - updates ionic

- fixes go mod vendor issue where build was broken in master branch
- fixes readme markdown for proper header tag

### CODE REVIEW

**Comment by struthj2** [comment]
my 2 cents on this: SPDX is a license aggregator/compliance format. It doesn't necessarily fit the use case for project import, as our definition of projects currently exist (project with valid source we can check out and examine or an artifact).

- possible improvements would be: moving this to ionic as a library function, putting this up for now so we can decide.
- including example spdx files for import and example usage .md file, this would be useful but not a pattern we currently have in this repo so I did not include them.

File: cmd/project.go [hunk 1]
```diff
@@ -186,3 +209,476 @@ var CreateProjectCmd = &cobra.Command{
...
+	p := projects.Project{}
+
+	fmt.Printf("Package %s\n", packageName)
+	pkgIDs, err := spdxlib.GetDescribedPackageIDs2_2(doc)
+	if err != nil {
+		fmt.Printf("Unable to get describe packages from SPDX document: %v\n", err)
+		return p, nil
+	}
+
+	// it does, so we'll go through each one
```
Comment (matthewkmayer) [inline cmd/project.go:434]: Let's clarify this comment with what "it" is and what it has. 👍

File: cmd/project.go [hunk 2]
```diff
@@ -186,3 +209,476 @@ var CreateProjectCmd = &cobra.Command{
...
+		Branch:      &branch,
+		Description: &description,
+		TeamID:      &teamID,
+		Type:        &ptype,
+		Source:      &source,
+		Name:        &name,
+		RulesetID:   &rulesetID,
+		Active:      active,
+		Monitor:     monitor,
+		//Aliases:     aliases,  // SPDX formats that may apply PackageOriginator, PackageSupplier, etc
```
Comment (matthewkmayer) [inline cmd/project.go:355]: What do you think about leaving this out by removing this line?

**Comment by matthewkmayer** [comment]
Let's get these commits squashed to meet our branching guidelines. 👍

File: cmd/project.go (hunk 1 above)
Comment (struthj2) [reply cmd/project.go]: updated in https://github.com/ion-channel/ionic/pull/454

File: ext/examples/SPDX_USAGE.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Usage
+
+```
+./ion-connect project create-project-spdx ./ext/examples/ex.spdx --spdx-version 2.1 --team-id <team-id>--ruleset-id <ruleset-id> --package-name hello-bin
```
Comment (energyvault462) [inline ext/examples/SPDX_USAGE.md:4]: nit: <team-id>--ruleset-id.    needs a space between > and --

**Review by energyvault462 — APPROVED** [review]

**Pull request merged**